src/views/Rules/ListRules/index.jsx

```diff
@@ -707,6 +716,7 @@ function ListRules(props) {
                 scrollToRow={scrollToRow}
                 rowHeight={getRowHeight}
                 rowCount={filteredRulesCount}
+                listRef={ruleListRef}
```

Comment:
Can this be renamed to `ref` instead of `listRef`? I _think_ `ref` is like `key` where the child component can't access that value and wont overwrite props.

```jsx
<ButtonComponent key={9999} />

function ButtonComponent(props) {
    console.log(props.key) // prints undefined
}
```

Reply:
Something like https://gist.github.com/mozbhearsum/3ed4f518e8c0d6e09ff3bc270ed31dca ?

I struggled a bit to make this work due to React things I don't quite understand. See https://github.com/facebook/flow/issues/7467, https://github.com/reactjs/rfcs/issues/103

Reply:
> Something like https://gist.github.com/mozbhearsum/3ed4f518e8c0d6e09ff3bc270ed31dca

Yes! Exactly that. Maybe I can help a bit:

`useRef`:
* is a hook
* common use case is to access a child methods
* can also be used as an instance variable in a class but that has nothing to do with what we're trying to do

In our case, we use `useRef` to have access to methods from the List component of react-virtualized.

`useImperativeHandle`:
* is a hook
* customizes the instance value (`ref` value) that is exposed to the parent when the parent decides to use `useRef` to have access to the child methods
* `useImperativeHandle` should be used with `forwardRef`

In our case, we let `VariableSizeList` take care of interacting with the internals of react-virtualized and expose a method via `useImperativeHandle` to ListRules that it can use.

`forwardRef`:
* Allows you to forward a `ref` prop down the React tree. React won’t pass some props down (e.g., `key` and `ref`) since they are considered somewhat special. `forwardRef` will have React allow it:

```jsx
// React will pass `ref` as a second parameter
function Child(props, ref) {

}
```

In our case, we use `forwardRef` to attach a method to that ref via `useImperativeHandle`.

Reply:
Ah, so `forwardRef` is really just a flag to say "allow the ref to be passed"?

Thank you for the detailed explanation, this is very helpful.

Reply:
Correct. No problem :)